### PR TITLE
installer: support multiple platforms and add platform check

### PR DESCRIPTION
**Pull request by KanjiMonster**
Add support for including support for multiple platforms, and do a check for support before starting the installation based on it.

Currently the installer is built around the assumption that it targets a single platform. To allow including multiple platforms, instead of creating a fixed `platform.conf` include multiple with the platform name in the path, so that we can include the appropriate `platform.conf` based on the ONIE information.

Since we then also have a easy way of finding out if the current platform is supported, use this to reject installation on unsupported platforms, and print a list of supported platforms.

Depends on https://github.com/bisdn/meta-open-network-linux/pull/69 to have the list of platforms exposed to the installer.

Depends on https://github.com/bisdn/meta-open-network-linux/pull/78 as a followup fix for the previous PR.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

File: recipes-core/images/onie-nos-installer.inc [hunk 1]
```diff
@@ -16,7 +16,7 @@ BISDN_ONIE_PLATFORM ?= "${@'${ONIE_PLATFORM}'.replace('_','-')}"
```
Comment (ideaship) [inline recipes-core/images/onie-nos-installer.inc:16]: In the commit message for "installer: flatten machines and use ONIE names",  s/a a /a /.

Also, we should document somewhere when it is _not_ sufficient to use a symlink for two similar platforms.

File: scripts/installer/install.sh [hunk 2]
```diff
@@ -253,7 +253,19 @@ create_bisdn_linux_msdos_partition()
...
 platform_check()
 {
-    /bin/true
+    onie_platform=$(onie-sysinfo -p)
+
+    if [ ! -d "./machine/${onie_platform}" ]; then
+      echo "ERROR: Unknown or unsupported platform: $onie_platform" >&2
+      echo "This image only supports the following platforms:" >&2
+      platforms="$(ls -1 ./machine)"
+      for plat in $flatforms; do
```
Comment (ideaship) [inline scripts/installer/install.sh:262]: flatforms?

File: scripts/installer/install.sh [hunk 3]
```diff
@@ -253,7 +253,19 @@ create_bisdn_linux_msdos_partition()
...
+         echo "    $platform"
```
Comment (ideaship) [inline scripts/installer/install.sh:263]: If the earlier lines go to stderr, this line should, too.

**Review by ideaship — CHANGES_REQUESTED** [review]

File: recipes-core/images/onie-nos-installer.inc (hunk 1 above)
Comment (KanjiMonster) [reply recipes-core/images/onie-nos-installer.inc:16]: Fixed, and reworded a bit to make it more clear why a symlink is sufficient here.

File: scripts/installer/install.sh (hunk 2 above)
Comment (KanjiMonster) [reply scripts/installer/install.sh]: Shorthand for "flattened platforms", should be obvious, no? ;p

File: scripts/installer/install.sh (hunk 3 above)
Comment (KanjiMonster) [reply scripts/installer/install.sh]: Also fixed, as well as the use of the correct variable here.

**Review by ideaship — APPROVED** [review]

**Pull request merged**